### PR TITLE
Testing PRS via QA nodes

### PR DESCRIPTION
**Pull request by chatman**
Integration test. 

Note: The existing test SolrCoreProxyTest wasn't behaving correctly with the MODIFYCOLLECTION, because the state.json wasn't found.

### CODE REVIEW

File: solr/core/src/test/org/apache/solr/core/FSPRSTest.java [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+
+      UpdateRequest ur = new UpdateRequest();
+      for(int i=0;i<10;i++) ur.add("id", ""+i);
+      ur.commit(cluster.getSolrClient(), COLL);
+
+      System.setProperty(CoreContainer.SOLR_QUERY_AGGREGATOR, "true");
+      JettySolrRunner qaJetty = cluster.startJettySolrRunner();
+      System.clearProperty(CoreContainer.SOLR_QUERY_AGGREGATOR);
+
+      try(HttpSolrClient client = (HttpSolrClient) qaJetty.newClient()){
```
Comment (hiteshk25) [inline solr/core/src/test/org/apache/solr/core/FSPRSTest.java:67]: Will this create CoreContainer instance for QueryAggregator node? Can we please add assert for it?

File: solr/core/src/test/org/apache/solr/core/FSPRSTest.java [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+
+
+      CollectionAdminRequest.SplitShard splitShard = CollectionAdminRequest.splitShard(COLL);
+      splitShard.setShardName("shard1");
+      NamedList<Object> response = splitShard.process(cluster.getSolrClient()).getResponse();
+      assertNotNull(response.get("success"));
+      for(int i= 0;i<100;i++) {
+        DocCollection docCollection = cluster.getSolrClient().getClusterStateProvider().getCollection(COLL);
+        int shardCount = docCollection.getSlices().size();
+        if(shardCount >=2) break;
```
Comment (hiteshk25) [inline solr/core/src/test/org/apache/solr/core/FSPRSTest.java:128]: Looks like we need to delete original shard after split. Also please add assert for Queryaggregator active slices to two?

Comment (noblepaul) [reply solr/core/src/test/org/apache/solr/core/FSPRSTest.java:128]: the parent shard is marked as `INACTIVE`
 
 Are you saying that without PRS, it used to DELETE the parent shard?

File: solr/core/src/test/org/apache/solr/core/FSPRSTest.java (hunk 1 above)
Comment (noblepaul) [reply solr/core/src/test/org/apache/solr/core/FSPRSTest.java:67]: done

**Review by hiteshk25 — APPROVED** [review]
LGTM(ran lucene solr tests: some failures)

```
119907    [junit4] Tests with failures [seed: 50131E35DF7A3EF] (first 10 out of 35):
119908    [junit4]   - org.apache.solr.cloud.LegacyCloudClusterPropTest.testCreateCollectionSwitchLegacyCloud
119909    [junit4]   - org.apache.solr.search.facet.TestJsonRangeFacets.testRangeFacetWithRangesInNewFormatDistrib
119910    [junit4]   - org.apache.solr.search.facet.TestJsonRangeFacets.testRangeFacetWithRanges
119911    [junit4]   - org.apache.solr.search.facet.TestJsonRangeFacets.testRangeFacetWithRangesInNewFormat
119912    [junit4]   - org.apache.solr.search.facet.TestJsonRangeFacets.testRangeOtherWhitebox
119913    [junit4]   - org.apache.solr.search.facet.TestJsonRangeFacets.testRangeFacetWithRangesDistrib
119914    [junit4]   - org.apache.solr.search.facet.TestJsonFacets.testRepeatedNumerics {p0=SMART}
119915    [junit4]   - org.apache.solr.search.facet.TestJsonFacets.testRepeatedNumerics {p0=STREAM}
119916    [junit4]   - org.apache.solr.search.facet.TestJsonFacets.testRepeatedNumerics {p0=DVHASH}
119917    [junit4]   - org.apache.solr.search.facet.TestJsonFacets.testRepeatedNumerics {p0=ENUM}
```

**Pull request merged**